Quality_Declaration_fastrtps.md

```diff
@@ -0,0 +1,137 @@
+
+18.  *Must have system, integration, and/or unit tests which cover all of the public API*
+    
+There are several [tests](https://github.com/eProsima/Fast-RTPS/tree/master/test) in the `fastrtps` repository, the unit tests folder, the communication, the profiling, and the realtime folders seem to be testing all the public API of the software.
+
+19.  *Must have code coverage, and a policy for changes*
+    
+Checking their [CI jobs](http://jenkins.eprosima.com:8080/), and the testing folders, there seems to be no automatic code coverage analysis being made or anything related to code coverage analysis.
+
+20.  *Performance tests, and performance regression policy*
+    
+Checking their [tests folder](https://github.com/eProsima/Fast-RTPS/tree/master/test), it is available a folder with performance tests, and also some [CI jobs](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Plots/) related to performance analysis. It is not possible to know whether there is a performance regression policy with the information from their repository.
+
+21.  *Linters and Static Analysis*
+    
+Based on the [CI jobs test results](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/lastSuccessfulBuild/testReport/), there seems there is no automatic static analysis being made on the code developed.
+
+
+### Dependencies:
+
+16.  Must not have direct runtime "ROS" dependencies which are not at the same level as the package in question ('Level N'), but…
+    
+
+`fastrtps` depends on the following packages:
+
+`libasio-dev` `libtinyxml2-dev` `fast-cdr` `foonathan_memory`
+
+The first two dependencies are suggested to be installed for Linux using apt package manager, which would pull them from the Debian upstream. This is a [list of advantages](https://wiki.debian.org/AdvantagesForUpstream) of using upstream packages, the fact that software and their patches are tested and the code is maintained available leads us to decide that these libraries as dependencies can be considered of Quality Level 1 as well.
+
+The other two are installed from the eProsima repositories as well, their current quality declarations can be found in [Quality Declaration `fast-cdr`](https://github.com/ros2/rmw_fastrtps/pull/360) and [Quality Declaration `foonathan_memory`](https://github.com/eProsima/foonathan_memory_vendor/pull/22)
+
+TO DO: Windows suggested method to install the dependencies is using [Chocolatey](https://chocolatey.org/) and the list of ROS2 dependencies [choco repo](https://github.com/ros2/choco-packages/releases/tag/2020-02-24). For Windows, the upstream argument won’t apply
+
+TO DO: Add link
+
+### Platform Support:
+
+23.  Must support all tier 1 platforms for ROS 2, as defined in [REP-2000](https://www.ros.org/reps/rep-2000.html#support-tiers)
+    
+As stated in their [repository page](https://github.com/eProsima/Fast-RTPS) the library is supported for Linux, Windows, and Mac, meaning that this requirement is met.
```

Comment:
Can you document that whether it supports the exact platforms (OS, version and architecture)?

Reply:
They don't have it stated in their [Supported platforms](https://github.com/eProsima/Fast-RTPS#supported-platforms) specifications. I'll change the wording to say "partially covered" as there is no claims about `aarch64` and specific OS.